DQMOffline/RecoB/plugins/BTagPerformanceAnalyzerOnData.cc

```diff
@@ -95,17 +101,22 @@ void BTagPerformanceAnalyzerOnData::bookHistograms(DQMStore::IBooker & ibook, ed
       
       // eta loop
       for ( int iEta = iEtaStart ; iEta < iEtaEnd ; ++iEta ) {
-	// pt loop
-	for ( int iPt = iPtStart ; iPt < iPtEnd ; ++iPt ) {
-	  
-	  const EtaPtBin& etaPtBin = getEtaPtBin(iEta, iPt);
-	  
-	  // Instantiate the genertic b tag plotter
-	  JetTagPlotter *jetTagPlotter = new JetTagPlotter(folderName, etaPtBin,
-							   iModule->getParameter<edm::ParameterSet>("parameters"), 0, false, ibook);
-	  binJetTagPlotters.at(iTag).push_back ( jetTagPlotter ) ;
-	  
-	}
+        // pt loop
+        for ( int iPt = iPtStart ; iPt < iPtEnd ; ++iPt ) {
+          
+          const EtaPtBin& etaPtBin = getEtaPtBin(iEta, iPt);
+          
+          // Instantiate the genertic b tag plotter
+          bool doDifferentialPlots = false;
+          double discrCut = -999.;
+          if (iModule->exists("differentialPlots") && iModule->getParameter<bool>("differentialPlots") == true) {
```

Comment:
this expensive check and the discrCut variable set can better be done outside of the loops

@swertz